x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/settings/TransportGetSecuritySettingsAction.java

```diff
@@ -0,0 +1,121 @@
+    GetSecuritySettingsAction.Response> {
+
+    @Inject
+    public TransportGetSecuritySettingsAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver
+    ) {
+        super(
+            GetSecuritySettingsAction.NAME,
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            GetSecuritySettingsAction.Request::new,
+            indexNameExpressionResolver,
+            GetSecuritySettingsAction.Response::new,
+            ThreadPool.Names.SAME
+        );
+    }
+
+    @Override
+    protected void masterOperation(
+        Task task,
+        GetSecuritySettingsAction.Request request,
+        ClusterState state,
+        ActionListener<GetSecuritySettingsAction.Response> listener
+    ) {
+        listener.onResponse(
+            new GetSecuritySettingsAction.Response(
+                getFilteredSettingsForIndex(SECURITY_MAIN_ALIAS, state),
+                getFilteredSettingsForIndex(SECURITY_TOKENS_ALIAS, state),
+                getFilteredSettingsForIndex(SECURITY_PROFILE_ALIAS, state)
+            )
+        );
+    }
+
+    /**
+     * Filters the settings to only those settable by the user (using the update watcher settings API).
+     */
+    private static Settings getFilteredSettingsForIndex(String indexName, ClusterState state) {
+        // Check the indices lookup to resolve the alias
+
+        return resolveConcreteIndex(indexName, state).map(idx -> state.metadata().index(idx))
+            .map(IndexMetadata::getSettings)
+            .map(settings -> {
+                Settings.Builder builder = Settings.builder();
+                for (String settingName : UpdateWatcherSettingsAction.ALLOWED_SETTING_KEYS) {
+                    if (settings.hasValue(settingName)) {
+                        builder.put(settingName, settings.get(settingName));
+                    }
+                }
+                return builder.build();
+            })
+            .orElse(Settings.EMPTY);
+    }
+
+    static Optional<Index> resolveConcreteIndex(String indexAbstractionName, ClusterState state) {
+        IndexAbstraction abstraction = state.metadata().getIndicesLookup().get(indexAbstractionName);
+        if (abstraction == null) {
+            return Optional.empty();
+        }
+        return Optional.ofNullable(abstraction.getWriteIndex());
+    }
+
+    @Override
+    protected ClusterBlockException checkBlock(GetSecuritySettingsAction.Request request, ClusterState state) {
+        ClusterBlockException globalBlock = state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_READ);
+        if (globalBlock != null) {
+            return globalBlock;
+        }
+
+        // Don't use the indexNameExpressionResolver here so we don't trigger a system index deprecation warning
```

Comment:
They're both hacks, but I'm not actually sure why what `TransportUpdateWatcherSettingsAction` does works to prevent the warning, which is emitted from `IndexNameExpressionResolver`, not anything `UpdateSettingsService` is doing.

(Also the way the Watcher does it will break if `.watches` is an alias - which I only caught because security's indices are aliased all the time - so I want to go and make Watcher use this way to resolve it)

Reply:
Also as I think about it maybe the right way to handle this is to put
```java
    @Override
    public boolean allowSystemIndexAccessByDefault() {
        return true;
    }
```

on the `Rest*SettingsAction` classes. Normally that's not what you want, because you largely should be doing operations on system indices in a system user security context which also prevents the warning, buuuuut I think this might be the oddball case where that's the semantically correct way of expressing what you want?

@williamrandolph do you have an opinion on this one, as the other system indices expert?

Reply:
Like normally the "correct" way to do this would be to use `OriginSettingClient` and execute transport actions with it to get that system user security context, but that's considerably less efficient than calling `UpdateSettingsService` directly.